connectors/sources/oracle.py

```diff
@@ -0,0 +1,60 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+"""SQLAlchemy source module is responsible to fetch documents from Oracle."""
+from urllib.parse import quote
+
+from sqlalchemy import create_engine
+
+from connectors.logger import logger
+from connectors.sources.generic_database import GenericBaseDataSource
+
+QUERIES = {
+    "PING": "SELECT 1+1 FROM DUAL",
+    "ALL_TABLE": "SELECT TABLE_NAME FROM all_tables where OWNER = '{user}'",
+    "TABLE_PRIMARY_KEY": "SELECT cols.column_name FROM all_constraints cons, all_cons_columns cols WHERE cols.table_name = '{table}' AND cons.constraint_type = 'P' AND cons.constraint_name = cols.constraint_name AND cons.owner = cols.owner ORDER BY cols.table_name, cols.position",
+    "TABLE_DATA": "SELECT * FROM {table}",
+    "TABLE_LAST_UPDATE_TIME": "SELECT SCN_TO_TIMESTAMP(MAX(ora_rowscn)) from {table}",
+    "TABLE_DATA_COUNT": "SELECT COUNT(*) FROM {table}",
+}
+
+
+class OracleDataSource(GenericBaseDataSource):
+    """Class to fetch documents from Oracle Server"""
```

Comment:
this docstring is used in the CLI, let's tweak the name, see https://github.com/elastic/connectors-python/pull/348